redisvl/extensions/cache/llm/langcache_api.py

```diff
@@ -0,0 +1,614 @@
+        """
+        if not 0 <= float(distance_threshold) <= 2:
+            raise ValueError("Distance threshold must be between 0 and 2")
+        self._distance_threshold = float(distance_threshold)
+
+    @property
+    def ttl(self) -> Optional[int]:
+        """Get the current TTL setting for cache entries."""
+        return self._ttl
+
+    def set_ttl(self, ttl: Optional[int] = None) -> None:
+        """Set the TTL for cache entries.
+
+        Args:
+            ttl: Time-to-live in seconds, or None to disable expiration.
+
+        Raises:
+            ValueError: If ttl is negative.
+        """
+        if ttl is not None and ttl < 0:
+            raise ValueError("TTL must be a positive integer or None")
+        self._ttl = ttl
+
+    def clear(self) -> None:
+        """Clear all entries from the cache while preserving the cache configuration."""
+        self._api.entries.delete_all(
+            cache_id=self._cache_id,
+            attributes={},
+            scope=(
+                self._entry_scope
+                if self._entry_scope is not None
+                else CacheEntryScope()
+            ),
+        )
+
+    async def aclear(self) -> None:
+        """Async clear all entries from the cache while preserving the cache configuration."""
+        await self._api.entries.delete_all_async(
+            cache_id=self._cache_id,
+            attributes={},
+            scope=(
+                self._entry_scope
+                if self._entry_scope is not None
+                else CacheEntryScope()
+            ),
+        )
+
+    def delete(self) -> None:
+        """Delete the cache and its index entirely."""
+        # First delete all entries
+        self.clear()
+        # Then delete the cache configuration
+        self._api.cache.delete(cache_id=self._cache_id)
+
+    async def adelete(self) -> None:
+        """Async delete the cache and its index entirely."""
+        # First delete all entries
+        await self.aclear()
+        # Then delete the cache configuration
+        await self._api.cache.delete_async(cache_id=self._cache_id)
+
+    def drop(self, ids: List[str]) -> None:
+        """Delete specific entries from the cache by their IDs.
+
+        Args:
+            ids: List of entry IDs to delete.
+        """
+        for entry_id in ids:
+            self._api.entries.delete(
+                cache_id=self._cache_id,
+                entry_id=entry_id,
+            )
+
+    async def adrop(self, ids: List[str]) -> None:
+        """Async delete specific entries from the cache by their IDs.
+
+        Args:
+            ids: List of entry IDs to delete.
+        """
+        for entry_id in ids:
+            await self._api.entries.delete_async(
+                cache_id=self._cache_id,
+                entry_id=entry_id,
+            )
+
+    def check(
+        self,
+        prompt: Optional[str] = None,
+        vector: Optional[List[float]] = None,
```

Comment:
should we make vector an optional **kwarg on the base model?

Reply:
Hmm. Isn't it already optional?